packages/types/lib/driver.ts

```diff
@@ -44,6 +44,19 @@ export interface SessionCommands {
   getSession(): Promise<SingularSessionData>;
 }
 
+export interface ExecuteCommands {
+  executeMethod(script: string, args: [Record<string, any>]|[]): Promise<any>;
+}
+
+export interface ExecuteOverloadDef {
+  command: string,
+  params?: {
+    required?: string[],
```

Comment:
we should be able to use this to define the type (sort of) of e.g. `soundzPlaySong(song, artist)`.  I can take a look later, but at minimum we will be able to determine that one or both are optional.  are these always `string` values?

Reply:
`required` and `optional` are lists of parameter *names*. so yes the names are always strings. their values could be anything.

so yeah if we want to update this we'll need to migrate to a schema system like the one you proposed.